lib/src/widgets/chat/ds_location_message_bubble.widget.dart

```diff
@@ -42,19 +43,77 @@ class DSLocationMessageBubble extends StatelessWidget {
 
     final lat = double.tryParse(latitude);
     final long = double.tryParse(longitude);
-
     final hasValidCoordinates = lat != null && long != null;
-    return GestureDetector(
-      onTap: () async {
-        final availableMaps = await MapLauncher.installedMaps;
+    List<AvailableMap>? availableMaps;
+
+    void initMaps() async {
+      availableMaps = await MapLauncher.installedMaps;
```

Comment:
These initializations (including ```async``` ones) should be done in ```initState``` method of a ```StatefulWidget```. That way, they'll execute just once per widget lifetime.

My suggestion is to transform this ```StatelessWidget``` into a ```StatefulWidget```, then do all this ```async``` processing in ```initState``` method and render ```availableMaps``` with ```FutureBuilder```.

Reply:
Done!